File: Client/AirsoftClient/src/app/models/products/guns/gunsViewModel.ts
import { AllGunViewModel } from "./allGunViewModel";

export interface GunsViewModel {
    allGuns: AllGunViewModel[],
    colors: string[],
    manufacturers: string[],
    dealers: string[],
    powers: number[],
    hasPreviousPage: boolean,
    hasNextPage: boolean,
    previousPageNumber: number,
    pageNumber: number,
    pagesCount: number,
    itemCount: number,
    itemsPerPage: number
}

File: Server/AirsoftShop.Controllers/Models/Products/GunsViewModel.cs
namespace AirsoftShop.Controllers.Models.Products;

using Common.Models;
using Services.Models.Products;

public class GunsViewModel : PagingModel
{
    public ICollection<GunViewServiceModel> AllGuns { get; set; }

    public ICollection<string> Colors { get; set; }

    public ICollection<string> Dealers { get; set; }

    public ICollection<string> Manufacturers { get; set; }

    public ICollection<double> Powers { get; set; }
}